### PR TITLE
[gha] Create release.yml and changelog.yml file

### PR DESCRIPTION
**Pull request by jjmerchante**
This PR adds a new GitHub Action to create the release for ELK. It generates a package, tests it, and publishes the release on PyPi and GitHub whenever a tag is generated and pushed.

This PR also includes a workflow to check if the changelog for the releases is included

It also updates `statsmodels` package in Poetry to 0.12.0 because 0.9.0 after a poetry build doesn't install the dependency numpy and fails.

This PR depends on https://github.com/chaoss/grimoirelab-github-actions/pull/1

### CODE REVIEW

**Review by vchrombie — CHANGES_REQUESTED** [review]
Thanks for opening the PR @jjmerchante 
This PR has a few conflicts, can you fix them and mark the PR as ready for review?

**Comment by vchrombie** [comment]
@jjmerchante we just have this PR left in the task list. It would be great if you can update the PR when you get some free time.
Thank you.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  push:
+    tags:
+      - '*.*.*'
+      - '*.*.*-*'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Build package using Poetry and store result."
```
Comment (vchrombie) [inline .github/workflows/release.yml:13]: small nit, can you remove the quotes from the `name`

**Review by vchrombie — COMMENTED** [review]
Very small change needed before we merge it, just to make the workflows consistent.

**Review by sduenas — APPROVED** [review]
LGTM, @vchrombie what do you think?

**Review by vchrombie — APPROVED** [review]
LGTM!

**Comment by vchrombie** [comment]
Thanks for the PR @jjmerchante.

**Pull request merged**